packages/frontend/src/components/accounts/CouldNotFindAccountModalWrapper.js

```diff
@@ -3,38 +3,91 @@ import { useDispatch } from 'react-redux';
 
 import {
     redirectTo,
-    refreshAccount
+    refreshAccount,
+    getLedgerPublicKey
 } from '../../redux/actions/account';
 import { showCustomAlert, clearGlobalAlert } from '../../redux/actions/status';
+import { setLedgerHdPath } from '../../utils/localStorage';
 import { getImplicitAccountIdFromSeedPhrase, getKeyPairFromSeedPhrase } from '../../utils/parseSeedPhrase';
-import { wallet } from '../../utils/wallet';
+import { wallet, setKeyMeta } from '../../utils/wallet';
 import { CouldNotFindAccountModal } from './CouldNotFindAccountModal';
 
-export function CouldNotFindAccountModalWrapper ({
+export function CouldNotFindAccountModalWrapper({
     isOpen,
     onClose,
-    seedPhrase
+    seedPhrase,
+    ledgerHdPath,
+    recoveryMethod
 }) {
     const dispatch = useDispatch();
+
+    const handleImportNonLedgerImplicitAccount = async () => {
+        const recoveryKeyPair = getKeyPairFromSeedPhrase(seedPhrase);
+        const implicitAccountId = getImplicitAccountIdFromSeedPhrase(seedPhrase);
+        try {
+            await wallet.importZeroBalanceAccount(implicitAccountId, recoveryKeyPair);
+            dispatch(refreshAccount());
+            dispatch(redirectTo('/'));
+            dispatch(clearGlobalAlert());
+        } catch (e) {
+            dispatch(showCustomAlert({
+                success: false,
+                messageCodeHeader: 'error',
+                messageCode: 'walletErrorCodes.recoverAccountSeedPhrase.errorNotAbleToImportAccount',
+                errorMessage: e.message
+            }));
+        }
+    };
+
+    const handleImportLedgerImplicitAccount = async () => {
+        try {
+            const ledgerPublicKey = await dispatch(getLedgerPublicKey(ledgerHdPath));
+            const implicitAccountId = Buffer.from(ledgerPublicKey.data).toString('hex');
+            const account = wallet.getAccountBasic(implicitAccountId);
+            try {
+                const accountState = await account.state();
+                if (accountState) {
+                    const errorMessage = `Implicit account ID ${implicitAccountId} derived from Ledger HD path ${ledgerHdPath} already exists but is not controlled by this Ledger device.`;
+                    // This could happen if user creates an implicit account with Ledger but then switches to seed phrase recovery, etc.
+                    console.log(errorMessage);
+                    dispatch(showCustomAlert({
+                        errorMessage: errorMessage,
+                        success: false,
+                        messageCodeHeader: 'error'
+                    }));
+                    return;
+                }
+            } catch (e) {
+                if (e.message.includes('does not exist while viewing')) {
+                    console.log('Ledger implicit Account ID does not exist on chain. Importing as zero balance account.');
```

Comment:
Same comment as above ⬆️

Reply:
I think it's worth keeping this prints to console because it'll be helpful for support in case of user error and for technical users